Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -705,7 +706,7 @@ mkDriverFee serviceName now startTime' endTime' merchantId driverId rideFare gov
         refundedAmount = Nothing,
         refundedAt = Nothing,
         refundedBy = Nothing,
-        vehicleCategory = currentVehicleCategory,
+        vehicleCategory = fromMaybe DVC.AUTO_CATEGORY currentVehicleCategory,
```

Comment:
What if vehicleCategory is null?

Reply:
it's already taken care of by variant in toTType of vehicle